eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/RegenerateUrlAliasesCommand.php

```diff
@@ -0,0 +1,1054 @@
+     */
+    protected $urlAliasGateway;
+
+    /**
+     * @var \Doctrine\DBAL\Connection
+     */
+    protected $connection;
+
+    /**
+     * @var int
+     */
+    protected $bulkCount;
+
+    /**
+     * @var \Symfony\Component\Console\Output\OutputInterface
+     */
+    protected $output;
+
+    protected $actionSet = [
+        'full' => true,
+        'autogenerate' => true,
+        'backup-custom' => true,
+        'restore-custom' => true,
+        'backup-global' => true,
+        'restore-global' => true,
+    ];
+
+    protected function configure()
+    {
+        $this
+            ->setName('ezpublish:regenerate:legacy_storage_url_aliases')
+            ->setDescription('Updates sort keys in configured Legacy Storage database')
+            ->addArgument(
+                'action',
+                InputArgument::REQUIRED,
+                'Action to perform, one of: full, autogenerate, backup-custom, restore-custom, backup-global, restore-global'
+            )
+            ->addArgument(
+                'bulk-count',
+                InputArgument::OPTIONAL,
+                'Number of items (Locations, URL aliases) processed at once',
+                50
+            )
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> regenerates URL aliases for Locations
+and migrates existing custom Location and global URL aliases to a separate database table. Separate
+table must be named '__migration_ezurlalias_ml' and should be created manually to be identical (but
+empty) as the existing table 'ezurlalias_ml' before the command is executed.
+
+After the script finishes, to complete migration the table should be renamed to 'ezurlalias_ml'
+manually.
+
+Using available options for 'action' argument, you can backup custom Location and global URL aliases
+separately and inspect them before restoring them to the migration table. They will be stored in
+backup tables named '__migration_backup_custom_alias' and '__migration_backup_global_alias' (created
+automatically).
+
+It is also possible to skip custom Location and global URL aliases altogether and regenerate only
+automatically created URL aliases for Locations (use 'autogenerate' action to achieve this).
+
+<error>During the script execution the database should not be modified.</error>
+
+Since this script can potentially run for a very long time, to avoid memory exhaustion run it in
+production environment using <info>--env=prod</info> switch.
+
+EOT
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->checkStorage();
+        $this->prepareDependencies($output);
+
+        $action = $input->getArgument('action');
+        $this->bulkCount = $input->getArgument('bulk-count');
+
+        if (!isset($this->actionSet[$action])) {
+            throw new RuntimeException("Action '{$action}' is not supported");
```

Comment:
Added in https://github.com/ezsystems/ezpublish-kernel/pull/1781/commits/cf42c749b2e6175797208f77581db40b9408c640